angr/state_plugins/heap/utils.py

```diff
@@ -0,0 +1,19 @@
+def concretize(x, solver, sym_handler):
+    """
+    For now a lot of naive concretization is done when handling heap metadata to keep things manageable. This idiom
+    showed up a lot as a result, so to reduce code repetition this function uses a callback to handle the one or two
+    operations that varied across invocations.
+
+    :param x: the item to be concretized
+    :param solver: the solver to evaluate the item with
+    :param sym_handler: the handler to be used when the item may take on more than one value
+    :returns: a concrete value for the item
+    """
+    if solver.symbolic(x):
+        sols = solver.eval_upto(x, 2)
```

Comment:
The kosher way of doing this these days is `state.solver.eval_one(x)` and then catching the SimSolverError if there's more than one

Reply:
Addressed [here](https://github.com/angr/angr/pull/1432/commits/1ed93117837470dcb9460f1bb8d109a7d1675b50).